addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -81,7 +81,21 @@ const PopupWidget = publicWidget.Widget.extend({
      */
     start: function () {
         this._popupAlreadyShown = !!getCookie(this.$el.attr('id'));
-        if (!this._popupAlreadyShown) {
+        // Check if every child element of the popup is conditionally hidden,
+        // and if so, never show an empty popup.
+        // config.device.isMobile is true if the device is <= SM, but the device
+        // visibility option uses < LG to hide on mobile. So compute it here.
+        const isMobile = config.device.size_class < config.device.SIZES.LG;
+        this._emptyPopup = [
```

Comment:
```suggestion
        const emptyPopup = [
```
Nitpicking but no need to add it on `this`